src/webapp/components/data-submission/UploadSample.tsx

```diff
@@ -1,92 +1,104 @@
+    const { api } = useAppContext();
+    const snackbar = useSnackbar();
 
-    const inputRef = useRef<HTMLInputElement>(null);
+    const [sampleFile, setSampleFile] = useState<File | null>(null);
+    const [isLoading, setIsLoading] = useState(false);
+    const sampleFileUploadRef = useRef<DropzoneRef>(null);
 
-    const files = fileList ? [...fileList] : [];
-
-    const updateFiles = (e: ChangeEvent<HTMLInputElement>) => {
-        setFileList(e.target.files);
-    };
+    const openFileUploadDialog = useCallback(async () => {
+        sampleFileUploadRef.current?.openDialog();
+    }, [sampleFileUploadRef]);
 
     const removeFiles = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
         event.preventDefault();
-        setFileList(null);
-    };
-
-    const selectFiles = () => {
-        if (inputRef.current != null) {
-            inputRef.current.click();
-        }
+        setSampleFile(null);
     };
 
-    const uploadFiles = () => {
-        if (!fileList) {
-            return;
-        }
-        // Create new FormData object and append files
-        const data = new FormData();
-        files.forEach((file, i) => {
-            data.append(`file-${i}`, file, file.name);
-        });
-
-        // TEST: Uploading the files using the fetch API to mock bin server
-        fetch("https://httpbin.org/post", {
-            method: "POST",
-            body: data,
-        })
-            .then(res => res.json())
-            .then(data => console.log(data))
-            .catch(err => console.error(err));
-    };
+    const sampleFileUpload = useCallback(
+        async (files: File[], rejections: FileRejection[]) => {
+            if (rejections.length > 0) {
+                snackbar.error(i18n.t("Multiple uploads not allowed, please select one file"));
+            } else {
+                const uploadedSample = files[0];
+                if (uploadedSample) {
+                    setIsLoading(true);
+                    setSampleFile(uploadedSample);
+                    try {
+                        const fileId = await uploadFile(api, uploadedSample);
+                        const submission = {
+                            id: generateUid(),
+                            batchId: "Dataset 1",
+                            call: "",
+                            countryCode: "",
+                            fileId,
+                            fileName: uploadedSample.name,
+                            fileType: "SAMPLE",
+                            inputLineNb: "",
+                            outputLineNb: "",
+                            module: "",
+                            period: "",
+                            specimens: [],
+                            status: "uploaded",
+                            submissionDate: new Date().toISOString(),
+                        };
+                        await saveToDataStore({ api, key: "submissions", object: submission });
```

Comment:
A component should invoke to a use case and this to a repository. the component should be agnostic of where the data is saved